### PR TITLE
Add a UI test for the settings page

### PR DESCRIPTION
**Pull request by karanbirsingh**
This UI test includes the following:
- spot-checking the application settings tab
- accessibility testing of the application settings tab
- spot-checking the about tab
- accessibility testing of the about tab
- changing a keyboard shortcut and validating it works
- validating the save and close button becomes enabled with changed options

Here's a video: https://dev.azure.com/accessibility-insights/Accessibility%20Insights/_build/results?buildId=1880&view=ms.vss-test-web.build-test-results-tab&runId=5176&resultId=100147&paneView=attachments

### CODE REVIEW

File: src/UITests/UILibrary/AIWinDriver.cs [hunk 1]
```diff
@@ -50,7 +50,9 @@ public int ScanAIWin(TestContext context)
             var result = scanner.Scan();
             if (result.ErrorCount > 0)
             {
-                context.AddResultFile(result.OutputFile.A11yTest);
+                var newPath = Path.Combine(outputPath, $"{fileName}.a11ytest");
```
Comment (DaveTryon) [inline src/UITests/UILibrary/AIWinDriver.cs]: FxCop may complain about this. The fix is:
```suggestion
                var newPath = Path.Combine(outputPath, Invariant($"{fileName}.a11ytest"));
```

and to add the following at the top of the file (line 9, maybe):

using static System.FormattableString;

File: src/UITests/UILibrary/AIWinDriver.cs [hunk 2]
```diff
@@ -62,6 +64,9 @@ public int ScanAIWin(TestContext context)
 
         public void ToggleHighlighter() => Session.FindElementByAccessibilityId(AutomationIDs.MainWinHighlightButton).Click();
 
+        public void GoToSettings() => Session.FindElementByAccessibilityId(AutomationIDs.SettingsButton).Click();
+
         public void Maximize() => Session.Manage().Window.Maximize();
+
```
Comment (DaveTryon) [inline src/UITests/UILibrary/AIWinDriver.cs]: Is this empty line intentional?

File: src/UITests/UILibrary/AIWinDriver.cs (hunk 1 above)
Comment (karanbirsingh) [reply src/UITests/UILibrary/AIWinDriver.cs]: updated, thanks for the suggestion

File: src/UITests/UILibrary/AIWinDriver.cs (hunk 2 above)
Comment (karanbirsingh) [reply src/UITests/UILibrary/AIWinDriver.cs]: mistake - I've removed it

File: src/AccessibilityInsights.SharedUx/Properties/AutomationIDs.cs [hunk 3]
```diff
@@ -13,5 +13,15 @@ public static class AutomationIDs
         public static string TelemetryDialogExitButton { get; } = nameof(TelemetryDialogExitButton);
```
Comment (karanbirsingh) [inline src/AccessibilityInsights.SharedUx/Properties/AutomationIDs.cs]: John and I will continue with this AutomationIDs strategy and reassess later in the feature whether we like it, want to group better, etc

File: src/AccessibilityInsights.SharedUx/Properties/AutomationIDs.cs [hunk 4]
```diff
@@ -13,5 +13,15 @@ public static class AutomationIDs
...
         public static string EventsDataGrid { get; } = nameof(EventsDataGrid);
         public static string EventModeControl { get; } = nameof(EventModeControl);
+        public static string SettingsButton { get; } = nameof(SettingsButton);
+        public static string SettingsApplicationTabItem { get; } = nameof(SettingsApplicationTabItem);
+        public static string SettingsAboutTabItem { get; } = nameof(SettingsAboutTabItem);
+        public static string ChannelComboBox { get; } = nameof(ChannelComboBox);
```
Comment (jalkire) [inline src/AccessibilityInsights.SharedUx/Properties/AutomationIDs.cs]: I've been prefixing these with (at least part of) the name of the control/class which contains them. As you mention, we probably want better organization in the future, so consider it a suggestion to use that naming style for now🙂

**Pull request merged**